include/envoy/network/connection.h

```diff
@@ -162,6 +162,16 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
    * Get the value set with setReadBufferLimit.
    */
   virtual uint32_t readBufferLimit() const PURE;
+
+  /**
+   * Set boolean telling if the connection is using an original destination address.
+   */
+  virtual void setUsingOriginalDst(bool using_original_dst) PURE;
```

Comment:
I would remove this from the interface. It should be passed as part of the Impl constructor.